File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        if (s.length() < 3) return s;
        
        StringBuilder result = new StringBuilder(s.substring(0, 2));
        int j = 2;
        
        for (int i = 2; i < s.length(); ++i) {
            if (s.charAt(i) != result.charAt(j - 1) || s.charAt(i) != result.charAt(j - 2)) {
                result.append(s.charAt(i));
                j++;
            }
        }
        
        return result.toString();
    }
}

File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        ret = []
        cnt = 1
        prev = None
        for c in s:
            if c == prev:
                cnt += 1
            else:
                cnt = 1
            prev = c
            if cnt < 3:
                ret.append(c)
        return ''.join(ret)
        